WcaOnRails/app/models/user_group.rb

```diff
@@ -77,15 +77,19 @@ def active_child_roles
   end
 
   def users
-    if self.group_type == UserGroup.group_types[:delegate_regions]
-      self.roles.map { |role| role[:user] }
-    else
-      self.roles.map(&:user)
-    end
+    self.roles.map { |role| role.is_a?(UserRole) ? role.user : role[:user] }
+  end
+
+  def child_users
+    self.child_roles.map { |role| role.is_a?(UserRole) ? role.user : role[:user] }
```

Comment:
But `role.child_roles` won't work since `child_roles` doesn't belong to roles, but belongs to groups. Did you mean `self.child_groups.flat_map(child_roles)`? If yes this is already done in `chile_roles`, so we can reuse it right?

Reply:
As far as I can see, everything boils down to calling `child_roles` at some point.

The current implementation of `def child_roles` only goes down one level: `child_groups.flat_map(&:roles)`. This is not necessarily _wrong_, but it might be unintuitive. We need to discern between three cases here:
1. Only consider roles that are *directly* associated with this group
2. Also consider roles that are related with *direct* children of this group
3. Also consider roles that are related with *any* children of this group (where "any" means it could be the child of a child of a child)

(1) is unambiguous. (2) and (3) could both be considered "child roles" depending on your interpretation. I don't have a strong opinion here on which is right and which is wrong, but I do have a strong opinion that we need to be _conscious_ about which approach works for us, and whether it makes sense to have `child_roles` recurse down or not.

Reply:
Agreed regarding the current ambiguity. I've modified the code in such a way that there is `direct_child_groups` and `all_child_groups`. `direct_child_groups` means the child groups which are directly connected, and `all_child_groups` will include the whole tree recursively. I also added some helper methods to fetch the roles and users of the group.